requests/models.py

```diff
@@ -403,10 +404,16 @@ def prepare_url(self, url, params):
     def prepare_headers(self, headers):
         """Prepares the given HTTP headers."""
 
+        self.headers = CaseInsensitiveDict()
         if headers:
-            self.headers = CaseInsensitiveDict((to_native_string(name), value) for name, value in headers.items())
-        else:
-            self.headers = CaseInsensitiveDict()
+            #Regex checks that string doesn't begin with whitespace or contain return characters.
+            #Otherwise, it checks if it's a blank string.
+            pat = re.compile(r'^\S[^\r\n]*$|^$')
+            for name, val in headers.items():
+                name, val = to_native_string(name), to_native_string(val)
```

Comment:
I'm a little bit worried about this, actually. `val` could be an opaque bytestring that doesn't decode well on Python 3. Let's change this to use a bytestring regex (prefixed with `rb`) and then elide the call to `to_native_string` on `val`.